pkg/embeddedcluster/monitor.go

```diff
@@ -0,0 +1,139 @@
+package embeddedcluster
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/replicatedhq/embedded-cluster-operator/api/v1beta1"
+	apptypes "github.com/replicatedhq/kots/pkg/app/types"
+	statetypes "github.com/replicatedhq/kots/pkg/appstate/types"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/store"
+	storetypes "github.com/replicatedhq/kots/pkg/store/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+// StartInstallationMonitor starts a goroutine that monitors the embedded cluster installation
+// and starts the upgrade process if necessary.
+func StartInstallationMonitor(ctx context.Context) error {
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		return fmt.Errorf("failed to get kubeclient: %w", err)
+	}
+	if isembedded, err := IsEmbeddedCluster(clientset); err != nil {
+		return fmt.Errorf("failed to check if embedded: %w", err)
+	} else if !isembedded {
+		return nil
+	}
+	mon := monitor{
+		store:     store.GetStore(),
+		clientset: clientset,
+	}
+	go mon.start(ctx)
+	return nil
+}
+
+// monitor is a struct that groups all methods needed to monitor the embedded cluster installation.
+type monitor struct {
+	store     store.Store
+	clientset *kubernetes.Clientset
+}
+
+// getApp returns the app deployed on top of the embedded cluster.
+func (m *monitor) getApp(ctx context.Context) (*apptypes.App, error) {
+	apps, err := m.store.ListInstalledApps()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list installed apps: %w", err)
+	} else if len(apps) == 0 {
+		return nil, nil
+	}
+	return apps[0], nil
```

Comment:
confirming that embedded clusters only support a single application? might be good to leave a comment about this here

Reply:
this has been changed to deploy the cluster type as part of the app deploy process

multiple apps with cluster specs WILL stomp on each other though